src/ucs/stats/stats.c

```diff
@@ -460,6 +477,184 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
+ * @param [in]  sel                       Whether or not to recursively iterate through the children.
+ * @param [in]  current_num_counters      Current number of counters iterated through (call with 0).
+ * @param [out] ucs_stats_aggregate_sum   The aggregate-sum of all counters of each class/type.
+ * @param [in]  num_stats_counters_max The number of elements in ucs_stats_aggregate_sum.
+ *
+ * Complexity: O(n)
+ *
+ * Returns: The total number of counters.
+ *
+ * Remarks: The function is recursive.
+ */
+static size_t ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
+                                             ucs_stats_children_sel_t sel,
+                                             size_t current_num_counters,
+                                             ucs_stats_counter_t *aggregate_sum,
+                                             size_t num_stats_counters_max)
+{
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    ucs_stats_node_t *temp_node;
+    ucs_stats_node_t *child;
+    uint32_t counter_index;
+    size_t n_counters = current_num_counters;
+    uint32_t cnt_agrgt_sum_index;
+    uint32_t counter_output_index;
+    uint32_t num_counters_filtered;
+
+    /* Filter output */
+    counter_output_index  = 0;
+    num_counters_filtered = ucs_popcount(filter_node->counters_bitmask);
+    cnt_agrgt_sum_index   = 0;
+    ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+        ucs_list_for_each(temp_node, &filter_node->type_list_head, type_list) {
+            cnt_agrgt_sum_index = counter_output_index +
+                                  ucs_stats_clsid_to_agrgt_sum_base_offset_get(
+                                          node, num_counters_filtered);
+
+            if (ucs_likely(cnt_agrgt_sum_index < num_stats_counters_max)) {
+                aggregate_sum[cnt_agrgt_sum_index] +=
+                        temp_node->counters[counter_index];
+            }
+        }
+        counter_output_index++;
+    }
+
+    n_counters += counter_output_index;
+
+    /* Children */
+    ucs_list_for_each(child, &node->children[sel], list) {
+        n_counters = ucs_stats_aggregate_sum_recurs(child, sel, n_counters,
+                                                    aggregate_sum,
+                                                    num_stats_counters_max);
+    }
+
+    return n_counters;
+}
+
+void ucs_stats_aggregate_sum(
+        ucs_stats_counter_t *aggregate_sum_counters,
+        size_t aggregate_sum_counters_size, size_t *n_counters_out,
+        const ucs_stats_aggrgt_sum_clsid_to_name_db_t **clsid_to_name_db)
+{
+    size_t n_counters            = 0;
+    ucs_stats_children_sel_t sel = UCS_STATS_ACTIVE_CHILDREN;
+
+    /* Zero output counters */
+    size_t total_size_bytes = sizeof(*aggregate_sum_counters) *
+                              aggregate_sum_counters_size;
+    memset(aggregate_sum_counters, 0x00, total_size_bytes);
```

Comment:
IMO better use for loop to init counters to 0, since it's counter value and not "raw" bytes
also, do we really need to init the whole buffer, or just the portion which will be used eventually?